packages/front-end/components/Experiment/EditMetricsForm.tsx

```diff
@@ -31,11 +31,23 @@ const EditMetricsForm: FC<{
   const { hasCommercialFeature } = useUser();
   const hasOverrideMetricsFeature = hasCommercialFeature("override-metrics");
 
-  const { metrics: metricDefinitions, getDatasourceById } = useDefinitions();
+  const {
+    metrics: metricDefinitions,
+    getDatasourceById,
+    getProjectById,
+  } = useDefinitions();
   const datasource = getDatasourceById(experiment.datasource);
-  const filteredMetrics = metricDefinitions.filter(
+  const project = getProjectById(experiment.project || "");
+  let filteredMetrics = metricDefinitions.filter(
     (m) => m.datasource === datasource?.id
   );
+  if (project) {
+    if (project?.metrics?.length) {
```

Comment:
Since you're using optional chaining (`?.`) you can consolidate to:
```suggestion
  if (project?.metrics?.length) {
```

Reply:
good catch